### ISSUE TITLE
service.name is not used when displaying services in Jaeger

### ISSUE THREAD

**Issue by MikeGoldsmith** [open-telemetry/opentelemetry-dotnet#589]
Discussed as part of [#567](https://github.com/open-telemetry/opentelemetry-dotnet/issues/567#issuecomment-608604953).

Configuring the service name in the JaegerExporter, the service name is not displayed in Jaeger correctly.

![image](https://user-images.githubusercontent.com/720635/78395055-65c10b00-75b2-11ea-98c6-d59e72da4c8b.png)

```csharp
services.AddOpenTelemetry(builder =>
{
    builder
        .SetSampler(new OpenTelemetry.Trace.Samplers.AlwaysOnSampler())
        .UseJaeger(o =>
        {

            o.ServiceName = "tracing-example-app";
            o.AgentHost = "tracing";
            o.AgentPort = 6831;
        })
        .SetResource(Resources.CreateServiceResource("tracing-example-app"))
        .AddRequestCollector()
        .AddDependencyCollector();
});
```

**Comment by brunomasantos** [comment]
Hi,

I was checking this issue and I was able to pinpoint the problem to this line: [https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerUdpBatcher.cs#L97](here)

Inverting the condition works as expected but dont know if it makes sense. Setting the Peer Service Name as the Span Service Name should be the default? Or it should be the tracer service name?
`var spanServiceName = this.Process.ServiceName ?? jaegerSpan.PeerServiceName;`

PeerService name is set here [https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerConversionExtensions.cs#L260](here). It matches the `http.host` key on the `PeerServiceKeyResolutionDictionary`.

Let me hear your thoughts about it, and if you like and can do the change.

**Comment by SergeyKanzhelev** [comment]
Peer name is about the service we call. Service name is the name of service reporting telemetry. Those are a different concepts. Are they mixed up in Jaeger exporter?

**Comment by brunomasantos** [comment]
From what I can see, on the reporting phase, the `SpanServiceName` is set with the Peer Service Name. I can see the tag `peer.service` being set but I don't know for sure is the concepts are mixed or not.

**Comment by longility** [comment]
I'm new to distributed tracing, opentelemetry, and jaeger so feel free to correct my understanding and help clarify as I'm trying to make sense of things from a very limited perspective.

1. In Jaeger, with the list of services, should we see the service name (of the reporting telemetry only) or also the peer name (calling service)? Is there a common practice or expectation?
2. From the issue, `localhost:32772` is the hostname of the service with the service name `tracing-example-app`. The reporting trace is collected because of the request collector. I would expect it should report the service name. As for dependency collectors, I would expect it to use the peer name as we can't determine the service name. Does that sound right?

So if we have service A calling service B, where service A and B have request collector and dependency collector, I think I should be expecting the following??

Reporting from A  |------------------------- A (service name) -------------------------|
Reporting from A..  |------------------------- B (peer name) -----------------------|
Reporting from B....  |---------------------- B (service name) --------------------|

**Comment by CodeBlanch** [comment]
Compared to Zipkin the Jaeger spec is a bit overcomplicated IMO. It requires you to send a batch for each service. If you don't, it thinks all the spans on a batch are internal and won't color them as different dependencies. This was added & working on #486 but I broke it on #524. Sorry! I just put in a fix and added a test case on #595 since it was already open in the area.

**Comment by MikeGoldsmith** [comment]
Thanks for the update @CodeBlanch 👍
